dev/planning-example.py

```diff
@@ -0,0 +1,265 @@
+
+# dates default to midnight if not otherwise specified
+day = obs.get_date("2015-05-01")
+
+# Can get more and more specific
+t1 = obs.get_date("2015-05-01 21:00")
+t1 = obs.get_date("2015-05-01 21:33:41")
+
+# default is current time
+cur_time = obs.get_date()
+
+# Dates are assumed to be in the observer's time zone unless a time zone
+# specifier is given
+t2_utc = obs.get_date("2015-05-01 21:33:41 UTC")
+
+"""
+We can ask about typical times of interest at this observing position
+Returned dates are assumed to be in the observer's time zone for convenience.
+The date parameter, if given, can be an astropy Time or a datetime instance.
+If no parameter is given, the current date/time is assumed.
+"""
+
+# sunset
+obs.sunset(date=day)
+
+# sunrise
+obs.sunrise(date=day)
+
+# moon rise
+obs.moon_rise(date=day)
+
+# moon set
+obs.moon_set(date=day)
+
+# evening (nautical) twilight
+obs.evening_twilight_12(date=day)
+
+# evening (civil) twilight
+obs.evening_twilight_18(date=day)
+
+# morning (nautical) twilight
+obs.morning_twilight_12(date=day)
+
+# morning (civil) twilight
+obs.morning_twilight_18(date=day)
+
+# what is the moon illumination?
+# returns a float, which is percentage of the moon illuminated
+obs.moon_illumination(date=t1)
+
+# what is the moon altitude and azimuth?
+# returns an altaz coordinate
+obs.moon_position(date=t1)
+
+# ==============
+# Target objects
+# ==============
+'''
+A target object defines a single observation target, with coordinates
+and an optional name.
+'''
+from astroplan.entity import SiderealTarget
+
+# Define a target.
+t1 = SiderealTarget(name='Polaris', ra='02:31:49.09', dec='+89:15:50.8')
+
+# initializing from astropy entities directly
+from astropy.coordinates import SkyCoord
+
+t1 = SiderealTarget(name='Polaris',
+               coord=SkyCoord('02h31m49.09s', '+89d15m50.8s', frame='icrs'))
+
+# Leaves scope for NonSiderealTarget, etc.
+
+# Convenience methods for looking up/constructing targets by name via
+# astroquery?
+
+
+# ================================
+# Condition objects, observability
+# ================================
+
+"""
+Q: Can I observe this target on the night of May 1, 2015 between 18:30
+and 05:30 HST for 30 minutes somewhere between 15 degrees and 89 degrees
+altitude?
+"""
+
+# first we define the conditions
+from astroplan.entity import Constraints
+
+# times can be passed in as strings (interpreted as for get_date()) or
+# as astropy Time or datetime objects
+cts = Constraints(time_start="2015-05-01 18:30", time_end="2015-05-02 05:30",
+                  alt_min=15.0*u.deg, alt_max=89.0*u.deg, duration=30*u.min)
+
+# define a target
+tgt = SiderealTarget(name='S5', ra='14:20:00.00', dec='48:00:00.00')
+
+# returns an object with information about the observability with these
+# constraints
+info = obs.observable(tgt, cts)
+
+'''
+attributes of return object:
+  observable: a boolean -- whether target is observable at desired constraints
```

Comment:
Perhaps this is out of scope or a bit high level, but how about an additional method to tack onto observable() that suggests alternate observations in the case that info = false (eg., time window that starts a bit earlier or later during the day chosen).  It could be called suggest, or suggest_time/suggest_alt/suggest_dur to break it up and look at changing just one constraint.  Could work something like info = obs.observable(tgt, cond).suggest_alt(tolerance).